src/lib/mina_base/account_update.ml

```diff
@@ -352,39 +374,76 @@ module Call_type = struct
   end = struct
     type variant = t
 
-    type 'bool t = { (* NB: call is implicit. *)
-                     is_delegate_call : 'bool }
+    type 'bool t =
+      { (* NB: call is implicit. *)
+        is_delegate_call : 'bool
+      ; is_blind_call : 'bool
+      }
     [@@deriving annot, hlist, fields]
 
-    let map ~f { is_delegate_call } = { is_delegate_call = f is_delegate_call }
+    let map ~f { is_delegate_call; is_blind_call } =
+      { is_delegate_call = f is_delegate_call; is_blind_call = f is_blind_call }
 
     let typ : _ Typ.t =
       let open Snark_params.Tick in
-      Typ.of_hlistable [ Boolean.typ ] ~var_to_hlist:to_hlist
-        ~var_of_hlist:of_hlist ~value_to_hlist:to_hlist ~value_of_hlist:of_hlist
+      let (Typ typ) =
+        Typ.of_hlistable
+          [ Boolean.typ; Boolean.typ ]
+          ~var_to_hlist:to_hlist ~var_of_hlist:of_hlist ~value_to_hlist:to_hlist
+          ~value_of_hlist:of_hlist
+      in
+      Typ
+        { typ with
+          check =
+            (fun ({ is_delegate_call; is_blind_call } as x) ->
+              let open Checked in
+              let%bind () = typ.check x in
+              let sum =
+                Field.Var.(add (is_delegate_call :> t) (is_blind_call :> t))
+              in
+              (* Assert boolean; we should really have a helper for this
+                 somewhere.
+              *)
+              let%bind sum_squared = Field.Checked.mul sum sum in
+              Field.Checked.Assert.equal sum sum_squared )
```

Comment:
I guess this `check` runs in addition to the check that the two fields are booleans individually?

Reply:
ah I guess that's the `typ.check x` above

Reply:
👍🏻